src/core_modules/capture-core/components/Pages/EnrollmentEditEvent/EnrollmentEditEventPage.container.js

```diff
@@ -46,7 +47,14 @@ export const EnrollmentEditEventPage = () => {
         dispatch(deleteEnrollment({ enrollmentId }));
     };
     const onGoBack = () => history.push(`/enrollment?${urlArguments({ orgUnitId, programId, teiId, enrollmentId })}`);
-    useEnrollment(teiId);
+    const enrollmentSite = useEnrollment(teiId).enrollment;
+    const { teiDisplayName } = useTeiDisplayName(teiId, programId);
+    const { trackedEntityName } = getScopeInfo(enrollmentSite.trackedEntityType);
+    const enrollmentsAsOptions = buildEnrollmentsAsOptions([enrollmentSite], programId);
+    const event = enrollmentSite?.events?.find(item => item.event === eventId);
```

Comment:
hey @simonadomnisoru how do we handle the case when `enrollmentSite` or `events` is undefined?

Reply:
Hey, 
Nice catch! I added checks for when the data is `undefined` in the [useEffect](https://github.com/dhis2/capture-app/pull/2022/files#diff-eeac4f86e764871045b77688d009436440c3f98f09af8a1c2b7d18b44d642105R58) bellow. 
Thanks!